src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -149,6 +148,8 @@ private[spark] object ArmadaClientApplication {
     s"armada-spark-app-id-${UUID.randomUUID().toString.replaceAll("-", "")}"
   private val DEFAULT_RUN_AS_USER = 185
 
+  private val gangId = Some(java.util.UUID.randomUUID.toString)
```

Comment:
This `gangId` is random, but within the lifetime of the JVM, it will be constant.

Meaning, you cannot create two client applications that have different gang ids.

Is this expected?